docs/docs/theme-api.md

```diff
@@ -50,90 +52,97 @@ module.exports = themeOptions => {
+Jeżeli chciałbyś, aby komponent `Header` robił coś innego (np. chciałbyś zmienić kolory, dodać dodatkową część, czy cokolwiek innego co możnaby zmodyfikować), wystarczy stworzyć plik w projekcie strony, która ten motyw wykorzystuje. Plik ten powinien znaleźć się w lokalizacji `src/gatsby-theme-amazing/components/header.js`. Teraz wystarczy wyeksportować dowolny komponent Reacta, a Gatsby użyje go zamiast odpowiadającego komponentu z motywu.
 
-> 💡 Note: you can shadow components from other themes using the same method. Read more about advanced applications in [latent shadowing](https://johno.com/latent-component-shadowing).
+> 💡 Można również używać w ten sposób komponentów z innych motywów. Dowiedz się więcej na temat zaawansowanych aplikacji w [latent shadowing](https://johno.com/latent-component-shadowing).
 
-### Extending
+### Rozszerzanie
 
-In the last section we talked about completely replacing one component with another. What if you want to make a smaller change that doesn't require copy/pasting the entire theme component into your own? You can take advantage of the ability to extend components.
+W poprzedniej sekcji opisaliśmy sposób nadpisywania komponentów własnymi. Ale co jeżeli chcielibyśmy tylko dodać coś do zapewnionych nam już komponentów, bez potrzeby kopiowania ich kodu do naszych? W tym celu, można skorzystać z możliwości **rozszerzania** komponentów.
 
-Taking the `Header` example from before, when you write your shadowing file at `src/gatsby-theme-amazing/components/header.js`, you can import the original component and re-export it as such, adding your own overridden prop to the component.
+Biorąc przykład `Header` z wcześniejszych sekcji, jeżeli stworzymy plik `src/gatsby-theme-amazing/components/header.js`, można zaimportować oryginalny komponent, a następnie wyeksportować go ponownie, nadpisując jakieś właściwości, czy korzystając z takich technik jak [Komponenty Wyższego Rzędu (HOC)](https://pl.reactjs.org/docs/higher-order-components.html)
 
 ```js
 import Header from "gatsby-theme-amazing/src/components/header"
 
-// these props are the same as the original component would get
+// Tutaj właściwości będą takie same, jakie otrzymałby oryginalny komponent, z tym że jedną z nich nadpisujemy/dodajemy ręcznie
 export default props => <Header {...props} myProp="true" />
 ```
 
-Taking this approach means that when you upgrade your theme later you can also take advantage of all the updates to the `Header` component because you haven't fully replaced it, just modified it.
+Idąc tą drogą, jeżeli później autor motywu go zaaktualizuje, to użytkownik nadal będzie mógł korzystać z wprowadzonych zmian, gdyż istniejący komponent będzie tylko rozszerzony, a nie wymieniony.
+
+### Jaka ścieżka powinna zostać wybrana, aby nadpisać plik?
+
```

Comment:
Tu chyba coś się pomieszało?

Reply:
Raczej kwestia Gita który miesza w tym miejscu diff. Poprzesuwały się akapity, bo opisałem to w punktach.

Reply:
U mnie linijka 97 to 94 z oryginału, dalej 99 odpowiada 96, a 101 do 105 odpowiada 98 :)

Reply:
A no to w porządku :)